### PR TITLE
External replication repair

### PR DESCRIPTION
**Pull request by teem0n**

### CODE REVIEW

File: internal/mysql/node.go [hunk 1]
```diff
@@ -510,10 +496,18 @@ func (n *Node) GetExternalReplicaStatus() (ReplicaStatus, error) {
...
-		"channel": n.config.ExternalReplicationChannel,
-	}, status, n.config.DBTimeout)
+
+	return n.ReplicaStatusWithTimeout(n.config.DBTimeout, n.config.ExternalReplicationChannel)
+}
+
+func (n *Node) ReplicaStatusWithTimeout(timeout time.Duration, channel string) (ReplicaStatus, error) {
+	query, status, err := n.GetVersionSlaveStatusQuery()
+	if err != nil {
+		return nil, nil
```
Comment (Fizic) [inline internal/mysql/node.go:506]: it is better to return the error further

File: internal/app/replication.go [hunk 2]
```diff
@@ -21,21 +21,22 @@ type ReplicationRepairState struct {
...
 	if replState.cooldownPassed(app.config.ReplicationRepairCooldown) {
-		delete(app.replRepairState, node.Host())
+		delete(app.replRepairState, key)
 	}
 }
 
-func (app *App) TryRepairReplication(node *mysql.Node, master string) {
-	replState := app.getOrCreateHostRepairState(node.Host())
+func (app *App) TryRepairReplication(node *mysql.Node, master string, channel string) {
+	replState := app.getOrCreateHostRepairState(app.makeReplStateKey(node, channel))
```
Comment (Fizic) [inline internal/app/replication.go:39]: maybe change the name of the function parameter name

File: internal/app/app.go [hunk 3]
```diff
@@ -1779,6 +1781,25 @@ func (app *App) repairCascadeNode(node *mysql.Node, clusterState map[string]*Nod
 	}
 }
 
+func (app *App) repairExternalReplication(masterNode *mysql.Node) {
```
Comment (Fizic) [inline internal/app/app.go:1784]: repairExternalReplication => TryRepairExternalReplication, for similarity with TryRepairReplication and maybe better move in separate file for external replication, like a common replication in replication.go

File: internal/app/app.go [hunk 4]
```diff
@@ -1779,6 +1781,25 @@ func (app *App) repairCascadeNode(node *mysql.Node, clusterState map[string]*Nod
...
+	extReplStatus, err := masterNode.GetExternalReplicaStatus()
+	if extReplStatus == nil {
```
Comment (Fizic) [inline internal/app/app.go:1786]: swap with if err != nil so as not to lose the error if it is

File: internal/app/replication.go [hunk 5]
```diff
@@ -57,12 +58,27 @@ func (app *App) TryRepairReplication(node *mysql.Node, master string) {
...
 	app.logger.Infof("repair: trying to repair replication using StartSlaveAlgorithm...")
+	if channel == app.config.ExternalReplicationChannel {
+		return node.StartExternalReplication()
+	}
 	return node.StartSlave()
 }
 
-func ResetSlaveAlgorithm(app *App, node *mysql.Node, master string) error {
+func ResetSlaveAlgorithm(app *App, node *mysql.Node, master string, channel string) error {
+	// TODO we don't want reset slave on external replication
```
Comment (Fizic) [inline internal/app/replication.go:77]: remove TODO?

**Review by Fizic — CHANGES_REQUESTED** [review]

File: internal/app/replication.go (hunk 5 above)
Comment (teem0n) [reply internal/app/replication.go:77]: No, at least not this time...
May be split algorithms in external/internal later.
Will write this in TODO

File: internal/app/app.go (hunk 3 above)
Comment (teem0n) [reply internal/app/app.go:1784]: No, this is an internal method, which is a wrapper for TryRepairReplication

**Review by Fizic — APPROVED** [review]

**Pull request merged**